### DISCUSSION TITLE
Add alll sidebar items to the default view list

### DISCUSSION THREAD

**Discussion by parfentjev** [navidrome/navidrome#3032] [Ideas]
In personal settings, we have the option to set the default view when opening the app. Currently, however, we are unable to set artists, songs, or the radio page as the default.

![image](https://github.com/navidrome/navidrome/assets/5242199/01401970-8468-4c9b-aa7c-a81b9966fbbd)

I suggest adding these menu items to the list. Personally, I typically either search for a specific song or shuffle through them; I'm not particularly interested in albums. Therefore, it would be more convenient for me to open the app on the songs page.

I found that the relevant code it tightly coupled to albums. In `ui/src/personal/SelectDefaultView.js`, only menu items relevant to albums are imported. It appears that there isn't a single entity encompassing all items in the sidebar. Additionally, the redirection doesn't seem to occur during the app initialization, but rather in the albums-related code found in `ui/src/album/AlbumList.js`.

As I'm not entirely comfortable tinkering with logic that I don't fully understand, I'm not submitting a pull request. However, I think that implementing this in the future would be beneficial.

Thank you in advance!

**Comment by deluan** [comment]
Apologies for how long this sat — thanks for the clear write-up and for pinpointing the album coupling in `SelectDefaultView`/`AlbumList`; it made this straightforward to pick up.

This is now addressed across two PRs:

- #5754 (merged) generalized the Default View selector beyond album lists and added **Artists**, **Songs**, and **Playlists** as options.
- #5801 builds on that to add **Radio** as well.

With both in, all the top-level lists you asked for are selectable as the default view. 🎉
